### PR TITLE
Fix support for custom filters without constructor in ApiFilter annotation

### PR DESCRIPTION
**Pull request by LouisPinsard**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets |  #3049 
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php [hunk 1]
```diff
@@ -77,6 +78,9 @@ private function createFilterDefinitions(\ReflectionClass $reflectionClass, Cont
             $definition->setAutowired(true);
 
             foreach ($arguments as $key => $value) {
+                if (null === $reflectionConstructor = $reflectionClass->getConstructor()) {
```
Comment (teohhanhui) [inline src/Bridge/Symfony/Bundle/DependencyInjection/Compiler/AnnotationFilterPass.php]: `$reflectionClass` could be `null` here. Also, this check should be done before the `foreach` loop.

File: tests/Fixtures/DummyFilterWithoutConstructor.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+namespace ApiPlatform\Core\Tests\Fixtures;
+
+use ApiPlatform\Core\Api\FilterInterface;
+
+/**
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class DummyFilterWithoutConstructor implements FilterInterface
+{
+    /**
```
Comment (teohhanhui) [inline tests/Fixtures/DummyFilterWithoutConstructor.php]: Do not copy the phpdoc here. Just use:
```php
    /**
     * {@inheritdoc}
     */
    public function getDescription(string $resourceClass): array
```

**Comment by LouisPinsard** [comment]
Hello @teohhanhui I modified the PR with your return if you can take a look !

**Comment by soyuka** [comment]
thanks @teohhanhui @LouisPinsard !

**Pull request merged**